systray_menu_linux.go

```diff
@@ -0,0 +1,205 @@
+package systray
+
+import (
+	"log"
+
+	"github.com/fyne-io/systray/internal/generated/menu"
+	"github.com/godbus/dbus/v5"
+	"github.com/godbus/dbus/v5/prop"
+)
+
+// SetIcon sets the icon of a menu item. Only works on macOS and Windows.
+// iconBytes should be the content of .ico/.jpg/.png
+func (item *MenuItem) SetIcon(iconBytes []byte) {
+}
+
+func (t *tray) GetLayout(parentId int32, recursionDepth int32, propertyNames []string) (revision uint32, layout menuLayout, err *dbus.Error) {
+	return 1, *instance.menu, nil
+}
+
+// GetGroupProperties is com.canonical.dbusmenu.GetGroupProperties method.
+func (t *tray) GetGroupProperties(ids []int32, propertyNames []string) (properties []struct {
+	V0 int32
+	V1 map[string]dbus.Variant
+}, err *dbus.Error) {
+	return
+}
+
+// GetProperty is com.canonical.dbusmenu.GetProperty method.
+func (t *tray) GetProperty(id int32, name string) (value dbus.Variant, err *dbus.Error) {
+	return
+}
+
+// Event is com.canonical.dbusmenu.Event method.
+func (t *tray) Event(id int32, eventId string, data dbus.Variant, timestamp uint32) (err *dbus.Error) {
+	if eventId == "clicked" {
+		item, ok := menuItems[uint32(id)]
+		if !ok {
+			log.Printf("Failed to look up clicked menu item")
+			return
+		}
+
+		item.ClickedCh <- struct{}{}
+	}
+	return
+}
+
+// EventGroup is com.canonical.dbusmenu.EventGroup method.
+func (t *tray) EventGroup(events []struct {
+	V0 int32
+	V1 string
+	V2 dbus.Variant
+	V3 uint32
+}) (idErrors []int32, err *dbus.Error) {
+	return
+}
+
+// AboutToShow is com.canonical.dbusmenu.AboutToShow method.
+func (t *tray) AboutToShow(id int32) (needUpdate bool, err *dbus.Error) {
+	return
+}
+
+// AboutToShowGroup is com.canonical.dbusmenu.AboutToShowGroup method.
+func (t *tray) AboutToShowGroup(ids []int32) (updatesNeeded []int32, idErrors []int32, err *dbus.Error) {
+	return
+}
+
+func createMenuPropSpec() map[string]map[string]*prop.Prop {
+	return map[string]map[string]*prop.Prop{
+		"com.canonical.dbusmenu": {
+			"Version": {
+				uint32(1),
+				false,
+				prop.EmitTrue,
+				nil,
+			},
+			"TextDirection": {
+				"ltr",
+				false,
+				prop.EmitTrue,
+				nil,
+			},
+			"Status": {
+				"active",
+				false,
+				prop.EmitTrue,
+				nil,
+			},
+			"IconThemePath": {
+				[]string{},
+				false,
+				prop.EmitTrue,
+				nil,
+			},
+		},
+	}
+}
+
+type menuLayout = struct {
+	V0 int32
+	V1 map[string]dbus.Variant
+	V2 []dbus.Variant
```

Comment:
What are V0, V1 and V2 purpose?

Reply:
Good question, added comments to clarify.
I cannot find a way to do better (like with nice variable names) as this is matching auto-generated code...